httpapi/httpapi.go

```diff
@@ -47,10 +55,26 @@ func sybilsHandler(hy *hydra.Hydra) func(w http.ResponseWriter, r *http.Request)
 // "/records/fetch" Receive a record and fetch it from the network, if available
 func recordFetchHandler(hy *hydra.Hydra) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
-		// TODO needs functionality to be implemented in libp2p
-		// See: https://discuss.libp2p.io/t/does-a-findproviders-replicate-the-provider-records-to-the-node-issuing-the-query/452
-		httpNotImplemented := 501
-		w.WriteHeader(httpNotImplemented)
+		cidStr := r.URL.Path[len("/records/fetch/"):]
+		cid, err := cid.Decode(cidStr)
+		if err != nil {
+			fmt.Printf("Received invalid CID, got %s\n", cidStr)
+
+			badRequest := 400
+			w.WriteHeader(badRequest)
+			return
+		}
+
+		ok := 200
+		w.WriteHeader(ok)
```

Comment:
Write 200 after first provider is found. If no providers are ever found write 404?